.github/workflows/limit-pr-files-changed.yml

```diff
@@ -0,0 +1,70 @@
+name: Enforce Max Lines Changed Per File
+
+on:
+  pull_request:
+    paths:
+      - '**/*.java'  # Monitor changes to Java files only
+
+jobs:
+  enforce-lines-changed:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Get PR Metadata
+        id: pr_details
+        uses: actions-ecosystem/action-get-PR@v2
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Check for Override Keyword
+        id: check_override
+        run: |
+          # Define the keyword for override
+          OVERRIDE_KEYWORD="override"
+          
+          # Check PR title and body for the keyword
+          if echo "${{ steps.pr_details.outputs.title }}${{ steps.pr_details.outputs.body }}" | grep -iq "$OVERRIDE_KEYWORD"; then
+            echo "Override keyword found. Skipping validation."
+            echo "override=true" >> $GITHUB_OUTPUT
+          else
+            echo "override=false" >> $GITHUB_OUTPUT
+          fi
+      - name: Calculate Lines Changed Per File
+        id: lines_changed_per_file
+        run: |
+          # Define the maximum allowed lines changed per file
+          MAX_LINES=500
+
+          # Get the diff of the PR and process each file
+          EXCEEDED=false
+          while IFS=$'\t' read -r added removed file; do
+          # Skip test files
+          if [[ "${file,,}" == *test.java ]]; then
```

Comment:
Does this mean test class files must end with the suffix `Test`? That does not seem right, since we don't follow this convention consistently enough. Can we instead rely on the full path needing to contain `src/main` in order to be included?

Reply:
Good idea! Let me update that  to check for full path, though developer should include "test" somewhere in the class name.